source/interprocedural_analyses/taint/taintConfiguration.ml

```diff
@@ -523,13 +522,49 @@ let parse source_jsons =
   }
 
 
-let validate { sources; sinks; features; _ } =
+let get_location ?encoding section label src =
+  let rec loop position section label section_hit d =
+    match Jsonm.decode d with
+    | `Lexeme (`String label_name) when String.equal label_name label ->
+        (if Int.equal section_hit 1 then
+           let (starting_row, starting_column), (ending_row, ending_column) =
+             Jsonm.decoded_range d
+           in
+           Log.error
+             "Entry for `%s` on lines %d - %d columns %d - %d"
+             label
+             starting_row
+             ending_row
+             starting_column
+             ending_column);
+        loop position section label section_hit d
+    | `Lexeme (`Name section_name) when String.equal section_name section ->
+        loop position section label (section_hit + 1) d
+    | `Lexeme (`Name another_section_name)
+      when List.exists
+             ~f:(String.equal another_section_name)
+             ["sources"; "sinks"; "features"; "rules"] ->
+        loop position section label (section_hit + 1) d
+    | `Lexeme _
+    | `Error _ ->
+        loop position section label section_hit d
+    | `End -> List.rev position
+    | `Await -> assert false
+  in
+  loop [] section label 0 (Jsonm.decoder ?encoding (`String src))
+
+
+let validate { sources; sinks; features; _ } configurations =
   let ensure_list_unique ~kind ~get_name elements =
     let seen = String.Hash_set.create () in
     let ensure_unique element =
       let element = get_name element in
-      if Hash_set.mem seen element then
-        failwith (Format.sprintf "Duplicate entry for %s: `%s`" kind element);
+      if Hash_set.mem seen element then (
+        Log.error "Duplicate entry for %s `%s` found:" kind element;
+        let section = kind ^ "s" in
+        let _ = List.map configurations ~f:(get_location section element) in
```

Comment:
We should make `get_location` side-effect free, return the locations and print them here.